src/main/java/emissary/output/DropOffUtil.java

```diff
@@ -997,6 +1009,31 @@ void extractUniqueFileExtensions(IBaseDataObject p) {
         }
     }
 
+    /**
+     * Checks the Original-Filename and FILE_ABSOLUTEPATH for the filename of the object. Returns a list with of the
+     * non-empty strings found in these fields. If nothing is found in either field, return an empty list.
+     *
+     * @param d The IBDO
+     * @return The list of filenames found in the field Original-Filename or FILE_ABSOLUTEPATH
+     */
+    public static List<String> getBestFilenames(IBaseDataObject d) {
+        String[] fieldsToTry = {ORIGINAL_FILENAME, FILE_ABSOLUTEPATH};
```

Comment:
We should probably make this a DEFAULT_FILENAME_FIELDS constant but allow the "working" list to be overwritten in the `configure(Configurator)` method.